### PR TITLE
adding multi control-plane node support

### LINKED ISSUE #10382: adding more control-plane nodes doesn't work
**Steps to reproduce the issue:** 

1. minikube start
2. minikube node add --control-plane
3. second node never succeeds

**Full output of failed command:** 
minikube --driver hyperkit --container-runtime crio start 
😄  minikube v1.17.0 on Darwin 11.1
✨  Using the hyperkit driver based on user configuration
👍  Starting control plane node minikube in cluster minikube
🔥  Creating hyperkit VM (CPUs=2, Memory=4000MB, Disk=20000MB) ...
🎁  Preparing Kubernetes v1.20.2 on CRI-O 1.19.0 ...
    ▪ Generating certificates and keys ...
    ▪ Booting up control plane ...
    ▪ Configuring RBAC rules ...
🔗  Configuring bridge CNI (Container Networking Interface) ...
🔎  Verifying Kubernetes components...
🌟  Enabled addons: storage-provisioner, default-storageclass

minikube node add --control-plane                         
😄  Adding node m02 to cluster minikube
❗  Cluster was created without any CNI, adding node to it might cause broken network.
E0205 17:29:35.683097   48700 register.go:137] unexpected first step: ""
👍  Starting node minikube in cluster minikube
🏃  Updating the running hyperkit "minikube" VM ...
E0205 17:29:37.406439   48700 register.go:137] unexpected first step: ""
🎁  Preparing Kubernetes v1.20.2 on CRI-O 1.19.0 ...

**Full output of `minikube start` command used, if not already included:**

**Optional: Full output of `minikube logs` command:**
<details>

</details>

### PR DESCRIPTION
**Pull request by michaelhenkel**
This patch make the --control-node argument in minikube node add work. minikube node add --control-plane now adds a new control-plane node

### CODE REVIEW

**Comment by michaelhenkel** [comment]
Can someone look into the PR?

**Comment by michaelhenkel** [comment]
Can some look into the PR?

File: pkg/minikube/bootstrapper/kubeadm/kubeadm.go [hunk 1]
```diff
@@ -752,6 +758,31 @@ func (k *Bootstrapper) GenerateToken(cc config.ClusterConfig) (string, error) {
...
+
+	if controlPlane {
+		conf := bsutil.KubeadmYamlPath
+		certKeyCmd := exec.Command("/bin/bash", "-c", fmt.Sprintf("%s init phase upload-certs --config %s --upload-certs", bsutil.InvokeKubeadm(cc.KubernetesConfig.KubernetesVersion), conf))
+		r, err = k.c.RunCmd(certKeyCmd)
+		if err != nil {
+			return "", errors.Wrap(err, "getting cert key")
+		}
+		certKey := r.Stdout.String()
+		var lines []string
```
Comment (tstromberg) [inline pkg/minikube/bootstrapper/kubeadm/kubeadm.go:770]: I believe this code could this be simplified to: `lines := strings.Split(r.Stdout.String(), "\n")`

Also, move the declaration of `lines` as close as possible to its first usage.

File: pkg/provision/provision.go [hunk 2]
```diff
@@ -246,6 +246,10 @@ CRIO_MINIKUBE_OPTIONS='{{ range .EngineOptions.InsecureRegistry }}--insecure-reg
 		return err
 	}
 
+	if _, err = p.SSHCommand("sudo systemctl restart crio"); err != nil {
```
Comment (tstromberg) [inline pkg/provision/provision.go:249]: This seems unrelated to the change. Can it be split out?

**Review by tstromberg — CHANGES_REQUESTED** [review]
PR looks great - just two minor comments and then it should be basically good to go for merging.

It would also be nice to see an integration test added, either as part of this PR or a follow-up, to ensure that we do not accidentally break this feature in the future.

File: pkg/provision/provision.go (hunk 2 above)
Comment (michaelhenkel) [reply pkg/provision/provision.go]: done

File: pkg/minikube/bootstrapper/kubeadm/kubeadm.go (hunk 1 above)
Comment (michaelhenkel) [reply pkg/minikube/bootstrapper/kubeadm/kubeadm.go]: done

**Comment by andriyDev** [comment]
Hi @michaelhenkel, would you mind rebasing this onto master?
